apps/designer/app/canvas/features/text-editor/interop.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  type ElementNode,
+  $getRoot,
+  $createTextNode,
+  $createParagraphNode,
+  TextNode,
+  LineBreakNode,
+  ParagraphNode,
+} from "lexical";
+import { $createLinkNode, LinkNode } from "@lexical/link";
+import type { ChildrenUpdates, Instance } from "@webstudio-is/react-sdk";
+import { createInstanceId } from "~/shared/tree-utils";
+
+// Map<nodeKey, Instance>
+export type Refs = Map<string, Instance>;
+
+const $writeUpdates = (
```

Comment:
I guess no that its a simple function, we could add a few tests (in case its simple enough to do, otherwise we have to postpone this for later)

Reply:
We will change this a lot when support nesting and other features.